### ISSUE TITLE
No Git info showing up

### ISSUE THREAD

**Issue by ejsc1** [pwr-Solaar/Solaar#1868]
**Information**
- Solaar version: 1.1.7+git1906-91b9a3a~ubuntu22.10.1
- Distribution: Ubuntu 22.10 + https://launchpad.net/~solaar-unifying/+archive/ubuntu/ppa?field.series_filter=kinetic
- Kernel version : Linux 6.1.0-060100rc5-generic x86_64 GNU/Linux
- Output of `solaar show`:

```
$ pip uninstall solaar
Found existing installation: solaar 1.1.7
Uninstalling solaar-1.1.7:
[...]

$ solaar show
No git dir found
Solaar version 1.1.7

[...]
```

**Comment by pfps** [comment]
The Ubuntu git PPA has not yet picked up the most recent commit.

https://launchpad.net/~solaar-unifying/+archive/ubuntu/ppa/+sourcepub/14159302/+listing-archive-extra

**Comment by pfps** [comment]
Yes, that would indicate that the most-recent PR has not been applied so git describe is being run in the current directory.   If the current directory is a git clone the git information for that directory will be picked up.  The most-recent PR runs the git describe inside the directory where the solaar executable is.

**Comment by pfps** [comment]
Odd.  Can you post the contents of lib/solaar/version.py from the package?

**Comment by pfps** [comment]
It looks as if the Ubuntu build process doesn't cd to the Solaar directory.

@muzena @rathann Do you know of a good way of getting the git commit into a file?  I'm trying to run `git describe` in setup.py but that isn't working in the Ubuntu build process.

**Comment by muzena** [comment]
@pfps 
git commands do not work in stupid PPA build system.
I had this problem in DevilutionX, only way is to extract git commit from ppa build path something like this in [DevilutionX](https://github.com/muzena/devilutionX/blob/debian_build_arm64_aarch64/Packaging/debian/rules)
Line 66 and 17 in rules file.
I will try to fix this in a few days.

**Comment by pfps** [comment]
Someone will have to figure out how to create a flatpak for Solaar, which appears to need help from the flatpak people.

**Comment by pfps** [comment]
It is unclear what exactly should be done.  I was trying to use `git describe` to get git information, which works when building or running from a clone of the git repository.   

If Ubuntu has a way of getting this information then I can add code to do that as well.

**Comment by pfps** [comment]
@muzena  Is it possible to run git before the repository is uploaded to the PPA?  If so then running `git describe --always > lib/solaar/commit` should put the commit information where Solaar expects it.  This should only be done for the PPA that tracks the git, not the one that only picks up releases.

**Comment by muzena** [comment]
@pfps 
`Is it possible to run git before the repository is uploaded to the PPA?`
No.
Only if you manually upload to PPA from PC.

**Comment by muzena** [comment]
@pfps 
Only way is like in [this post](https://github.com/pwr-Solaar/Solaar/issues/1868#issuecomment-1323900719) for git version and commit to extract and replace with sed command something like this in [Linux Mint rule file](https://github.com/linuxmint/mintupdate/blob/master/debian/rules)

**Comment by pfps** [comment]
@muzena What is the process that extracts Solaar from github and where does that run?

**Comment by pfps** [comment]
@ejsc1 The problem is that the Python setup code is being run in a directory that does not have git information.  This information was available at some point, because it shows up in the PPA, so it is just figuring out the best way to get the information.

**Comment by pfps** [comment]
PR #1880 is an attempt to get the git commit from the Ubuntu PPA directory name.  It's not guaranteed to work, but I'm going to push it so that the PPA will pick it up and it will either work or provide some information that can be used to fix it.

**Comment by pfps** [comment]
@muzena In https://launchpadlibrarian.net/635811255/buildlog_ubuntu-jammy-amd64.solaar_1.1.7+git1911-0547b18~ubuntu22.04.1_BUILDING.txt.gz the git commit ID doesn't appear to show up in the directory during setup.py.  What is going wrong?

**Comment by muzena** [comment]
@pfps 
From [build log](https://launchpadlibrarian.net/636215458/buildlog_ubuntu-focal-amd64.solaar_1.1.7+git1914-0c7a9dd~ubuntu20.04.1_BUILDING.txt.gz) build path (**build/solaar-MAZXir/solaar-1.1.7+git1914**) is in lines:



Local sources
-------------

solaar_1.1.7+git1914-0c7a9dd~ubuntu20.04.1.dsc exists in .; copying to chroot
I: NOTICE: Log filtering will replace 'build/solaar-MAZXir/solaar-1.1.7+git1914' with '<<PKGBUILDDIR>>'
I: NOTICE: Log filtering will replace 'build/solaar-MAZXir' with '<<BUILDDIR>>'

**Comment by pfps** [comment]
@ejsc1 There should be files with names version and commit in the distribution directories.  Where are they and what are their contents?

**Comment by pfps** [comment]
@muzena Thanks.

**Comment by pfps** [comment]
The Solaar code is somewhere.  I don't know exactly where but you should be able to `find` it via `find /usr -name solaar`, but you may have to use something other than /usr.   One of the results should be the directory where the Solaar sources are.  Then look in that directory for files named version and commit and post their contents.

**Comment by pfps** [comment]
One of these directories, most likely /usr/share/solaar/lib/solaar will have a file named version in it, and probably a file named commit.  Post the contents of these files.

**Comment by pfps** [comment]
Thanks.  I was trying to do something smart and it does not appear to have worked.  I'll put in a change and hopefully things will work for the next build.

**Comment by pfps** [comment]
Yeah.  A stupid bug in setup.py that I just fixed.

**Comment by muzena** [comment]
Yo can [start it manually](https://code.launchpad.net/~solaar-unifying/+recipe/solaar-unifying-git), by clicking on "[Request build(s)](https://code.launchpad.net/~solaar-unifying/+recipe/solaar-unifying-git/+request-builds)"

**Comment by pfps** [comment]
@muzena There is some lag between the time updates are made on GitHub and the time they are available to the PPA build process.  Do you know how long that is?

**Comment by muzena** [comment]
@pfps 
Lanchpad scans github every 5 hours for changes, but you can manually import clicking on [Import now button](https://code.launchpad.net/~solaar-unifying/solaar-unifying/+git/solaar-unifying), and then after import finishes click on  [Request build(s)](https://code.launchpad.net/~solaar-unifying/+recipe/solaar-unifying-git) to manually start building.

**Comment by pfps** [comment]
Fixed by #1901
